File: cp_nim/src/bin/abc434-b-bird-watching.nim
# Created by Ayush Biswas at 2026/07/18 18:43
# https://atcoder.jp/contests/adt_easy_20260617_1/tasks/abc434_b
import ../lib/cpio

# @code begin

import sequtils

var r = newReader()

let
  (n, m) = r[int, int]
  birds = r[seq[seq[int]], n]

var weights = newSeq[int](m)
var counts = newSeq[int](n)

for bird in birds:
  weights[bird[0] - 1] += bird[1]
  counts[bird[0] - 1] += 1

for i in 0 ..< m:
  echo weights[i] / counts[i]

# @code end

File: cp_rust/src/bin/abc434-b-bird-watching.rs
// Created by Ayush Biswas at 2026/07/07 20:36
// https://atcoder.jp/contests/adt_easy_20260617_1/tasks/abc434_b
use cp_lib::*;

// @code begin
use cpio::*;

sol! {
    fn solution(
        [n, m]: [usize; 2],
        abs: [[u32; 2]; n]
    ) -> Lines<f64> {
        let mut sums = vec![0; m + 1];
        let mut counts = vec![0; m + 1];
        for [a, b] in abs {
            counts[a as usize] += 1;
            sums[a as usize] += b;
        }
        let mut res = vec![];
        for i in 1..=m {
            res.push(f64::from(sums[i])/f64::from(counts[i]));
        }
        res.into()
    }
}
// @code end
